### DISCUSSION TITLE
Home assistant doesn't autodiscover

### DISCUSSION THREAD

**Discussion by DawidWetzler** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#769] [Q&A]
Hi there,

I just discovered this project two days ago and gave it a immediate go and who would have though I got a smart remote for the pump now! Big thanks to the author and all the contributors!

What I'm trying to do now is to integrate it somehow to home assistant. I've read that it does support home assistant auto discovery but somehow it sadly isn't working for me.

My ESP D1 Mini says it's connected to the MQTT Server.
<img width="241" alt="Screenshot 2024-07-19 at 14 43 18" src="https://github.com/user-attachments/assets/e40531e0-b5ac-421f-8dc4-8b859c071e13">

MQTT Explorer proves it:
<img width="291" alt="Screenshot 2024-07-19 at 14 44 02" src="https://github.com/user-attachments/assets/9ebad4fd-0a28-4439-8c92-4e60f582726f">

and of course I can see home assistant and zigbee2mqtt reporting to the same server

Home assistant MQTT is receiving the messages:
<img width="608" alt="Screenshot 2024-07-19 at 14 44 25" src="https://github.com/user-attachments/assets/8e2c3f68-6247-47ef-b6f0-63db347d220a">

The configuration of the ESP for S100101 looks like this:
<img width="427" alt="Screenshot 2024-07-19 at 14 45 06" src="https://github.com/user-attachments/assets/998a4371-fd4e-4115-83d7-2660b4d35fc0">

The configuration of the MQTT looks like this:
<img width="411" alt="Screenshot 2024-07-19 at 14 46 12" src="https://github.com/user-attachments/assets/9fc8589c-a7e1-49ec-8402-9feb725d2b3b">

Additionally I tried:
- restarting esp multiple times
- disconnecting the power supply
- playing around with the settings to let it report something
- changing the settings (to other pump models)
- deleting the whole entry with mqttexplorer to "re-add" it
- disconnecting network completely to let it reconnect

If somebody has an idea or clue what I might be doing wrong I'd really appreciate your support :)
Thanks in advance.

@edit1: I just realised that there was a new release
the firmware version is now:

```Firmware: 2024-07-19-0816```

**Comment by visualapproach** [comment]
Hi, it takes a few minutes before it sends the configuration to HA. You should see topics like homeassistant/binary_sensor/layzspa_connection8079969/config if it has sent auto discovery messages.
You should see the device in ha under settings, devices, mqtt

Glad you like the project!

**Comment by DawidWetzler** [comment]
Thank you for your quick response!

It's connected to the mqtt and network since yesterday.

I do see such messages in MQTT Explorer.
<img width="430" alt="Screenshot 2024-07-19 at 15 29 07" src="https://github.com/user-attachments/assets/0c14e46f-17a5-4897-b341-e2a8069cec38">

example topic: homeassistant/binary_sensor/layzspa_connection5359082

Although still it's not being discovered in home assistant.
<img width="909" alt="Screenshot 2024-07-19 at 15 33 46" src="https://github.com/user-attachments/assets/4e8dd89a-b1b2-48bd-b400-639a942370ba">

<img width="329" alt="Screenshot 2024-07-19 at 15 35 38" src="https://github.com/user-attachments/assets/3497f402-6d47-46fb-b787-cfc4b508e90b">



Edit: 1
I tried deleting everything in this regards and it actually republished everything but nothing is being seen by home assistant.

**Reply by visualapproach** [reply]
What if you restart ha? And maybe after that restart esp.
Do you have a recent version of ha?

**Reply by DawidWetzler** [reply]
I just now updated from 2024.7.0 to 2024.7.2 to try if anything changes.

Still nothing is visible.

**Reply by visualapproach** [reply]
And you have enabled autodiscovery in ha?

**Reply by DawidWetzler** [reply]
MQTT says it's enabled and also the zigbee devices are being auto discovered.

<img width="580" alt="Screenshot 2024-07-19 at 16 15 12" src="https://github.com/user-attachments/assets/82e5a2a5-2ed1-457d-8a1c-674364c5e1d1">

Edit:
- changed user for mqtt (nothing changed)
- tried updating mqtt for whatever reason (nothing chanted)

I'm running out of ideas 😆

**Reply by DawidWetzler** [reply]
Okay, so it finally connected and got discovered.

I'm not sure what it did but it's now there.

Everything it doubled for whatever reason but it's there :D 

Thank you very much for your support!

<img width="322" alt="Screenshot 2024-07-19 at 16 50 53" src="https://github.com/user-attachments/assets/cea77082-3f11-43e2-807c-d8aaf2c2be39">

**Reply by visualapproach** [reply]
Strange! But if you dare, remove the mqtt latzspa device in ha and then restart esp and wait 11 minutes. Then it should come back as it should. It turns up in just a minute or two here but to be safe.
